src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -103,6 +107,19 @@
     private final Cache<String, ActiveDirectoryGroupDetails> groupCache;
 
     /**
+     * The threadPool to update the cache on background
+     */
+    private final ExecutorService threadPoolExecutor =
```

Comment:
might make more sense to have this as part of the descriptor and get initialized on first use and shut down from a `@Terminator`